lldb/source/Plugins/SymbolLocator/Debuginfod/SymbolLocatorDebuginfod.cpp

```diff
@@ -114,29 +177,85 @@ SymbolLocator *SymbolLocatorDebuginfod::CreateInstance() {
 
 static std::optional<FileSpec> GetFileForModule(
     const ModuleSpec &module_spec,
-    std::function<llvm::Expected<std::string>(llvm::object::BuildIDRef)>
-        PullFromServer) {
+    std::function<llvm::StringRef(llvm::object::BuildID)> UrlBuilder,
+    bool force_lookup) {
+  PluginProperties &plugin_props = GetGlobalPluginProperties();
+  llvm::Expected<llvm::StringRef> CacheDirectoryPathOrErr =
+      plugin_props.GetCachePath();
+  // We really do require a cache location to store stuff.
+  if (!CacheDirectoryPathOrErr)
+    return {};
```

Comment:
Will this have a default value from the default debuginfod cache location? Or does this require us to always set a cache directory in the settings?

Reply:
Yes. It defaults to the debuginfod cache location, but can be overridden.